### PR TITLE
Added a post about guarding/compiling profiles

### PR DESCRIPTION
**Pull request by matejak**

### CODE REVIEW

File: _posts/2020-03-06-profile_guard.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+author: Matěj Týč
+---
+
+# Stay on Top of Your Profile
+
+## The problem: How is my profile?
+
+Let's consider the obvious question "how do you feel about your profile?".
+The answer certainly depends on the size of the profile in question - so how many rules and other settings define a profile?
+That's not such an easy question to answer either, but profiles based on policies have typically between one to four hundreds of rules selected.
```
Comment (jan-cerny) [inline _posts/2020-03-06-profile_guard.md]:
```suggestion
That's not such an easy question to answer either, but profiles based on policies have typically up to four hundreds of rules selected.
```

File: _posts/2020-03-06-profile_guard.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+Answering the original question about profile's state is made more tricky due to how ComplianceAsCode build system gives content authors space when specifying profiles.
+ComplianceAsCode uses the YAML format for profile definitions, so profile creators often group rules by policy requirements, and add a lot of comments why a specific rule is selected by the profile.
+As a result, profile definitions can end up having hundreds of lines.
+When you consider the ability of profiles to extend an existing profile, it's really difficult to keep track of profile composition over time.
+Ultimately, one can analyze the datastream and examine selections there, but one could object to that, as the build system at one point knows what's selected and what's not.
```
Comment (jan-cerny) [inline _posts/2020-03-06-profile_guard.md]: I would reword this to emphasise the build system.

File: _posts/2020-03-06-profile_guard.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+## The solution: Build system tells you
+
+The build system recently gained the ability to analyze profile definitions at build-time, and produce a "compiled profile" file - a YAML file that contains final rule selection, variable assignments, and rule refinements, all of that sorted lexicographicaly.
```
Comment (jan-cerny) [inline _posts/2020-03-06-profile_guard.md]:
```suggestion
The build system recently gained the ability to analyze profile definitions at build-time. It produces a "compiled profile" file - a YAML file that contains final rule selection, variable assignments, and rule refinements, all of that sorted lexicographicaly.
```

File: _posts/2020-03-06-profile_guard.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+A problem that our group has been facing in the past was an unintended introduction of profile changes.
+Imagine that you decide to restructure your profile definition by regrouping selections and adding a couple of comments - the resulting diff can easily grow over a hundred lines, and humans have no chance of reviewing the pull request properly, without losing their mind.
+Imagine also a situation when a profile you build upon changes, and you don't want those changes to propagate to your profile.
+
+Along with the introduction of compiled profiles, we have switched on an optional test - it is now possible to compare profile contents to reference contents.
+This basically means that a profile tested this way is specified twice in the project - there is its author-friendly definition in `<product>/profiles/<name>.profile`, and if there may be its computer-friendly, compiled definition at `tests/data/profile_stability/<product>/<name>.profile`.
+
+If the product is built and the test is executed, test references are compared to their corresponding compiled build artifacts.
+If selections differ, the test fails, and a helpful message is displayed.
```
Comment (jan-cerny) [inline _posts/2020-03-06-profile_guard.md]: It would be nice to have a picture here.

File: _posts/2020-03-06-profile_guard.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+The build system recently gained the ability to analyze profile definitions at build-time, and produce a "compiled profile" file - a YAML file that contains final rule selection, variable assignments, and rule refinements, all of that sorted lexicographicaly.
+In other words, profile extensions, blacklisting and overrides are resolved, comments are dropped, and selections ordering is normalized.
+
+For example, if you would like to know what is the real contents of the RHEL8 STIG profile, you don't look to `rhel8/profiles/stig.profile`.
+Instead, you build the RHEL8 datastream, and check out the `build/rhel8/profiles/stig.profile` build artifact.
+
+## The improvement: Stability test
+
+A problem that our group has been facing in the past was an unintended introduction of profile changes.
+Imagine that you decide to restructure your profile definition by regrouping selections and adding a couple of comments - the resulting diff can easily grow over a hundred lines, and humans have no chance of reviewing the pull request properly, without losing their mind.
```
Comment (jan-cerny) [inline _posts/2020-03-06-profile_guard.md]: "losing their mind" feels too strong

File: _posts/2020-03-06-profile_guard.md (hunk 2 above)
Comment (matejak) [reply _posts/2020-03-06-profile_guard.md]: I don't know how to address this.

**Comment by matejak** [comment]
I have updated the text and supplied it with some examples.

File: _posts/2020-03-06-profile_guard.md (hunk 2 above)
Comment (jan-cerny) [reply _posts/2020-03-06-profile_guard.md]:
```suggestion
Ultimately, one can analyze the datastream and examine selections there. But, the build system knows what's selected and what's not. We can use the build system to show this information instead of analysing a complex XML file.
```

**Comment by jan-cerny** [comment]
@matejak please move the conclusion to the end of the article

**Comment by matejak** [comment]
Done.
